client/state/jetpack/credentials/reducer.js

```diff
@@ -71,9 +71,26 @@ export const testRequestStatus = keyedReducer( 'siteId', ( state, { type, role }
 		case JETPACK_CREDENTIALS_TEST:
 			return {
 				...state,
-				[ role ]: 'pending',
+				[ role ]: true,
 			};
 
+		case JETPACK_CREDENTIALS_TEST_VALID:
+		case JETPACK_CREDENTIALS_TEST_INVALID:
+			return {
+				...state,
+				[ role ]: false,
+			};
+	}
+
+	return state;
+} );
+
+export const testStatus = keyedReducer( 'siteId', ( state, { type, role } ) => {
+	switch ( type ) {
+		// Do not affect current state if executing a new test
+		case JETPACK_CREDENTIALS_TEST:
```

Comment:
This should not be necessary, since the `type` will not match any of the existing `case`s. We should be safe to remove it.

Reply:
Yeah, you are right. I added it while testing because I did forget to `return $state` at the end, then I added it back but never fixed this. Working on it :)